### PR TITLE
update icons to ts enum version

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

`Pages` folder Icon imports updated to TS version

Update imports from `'../../component-library/component-name/deprecated';` to `'../../component-library'`
Update component instances from using JS consts to enums

* Part of #18492 

## Screenshots/Screencaps

See comments

## Manual Testing Steps
regex search `ICON_NAMES` in the `ui/pages` folder

Screenshot 2023-04-18 at 11 44 48 AM

`yarn lint`
`yarn jest ui/pages`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/add-nft/add-nft.js [hunk 1]
```diff
@@ -37,10 +37,10 @@ import {
   MetaMetricsTokenEventSource,
 } from '../../../shared/constants/metametrics';
 import {
-  ICON_NAMES,
-  ICON_SIZES,
-} from '../../components/component-library/icon/deprecated';
-import { ButtonIcon } from '../../components/component-library/button-icon/deprecated';
+  ButtonIcon,
+  IconName,
+  IconSize,
```
Comment (NidhiKJha) [inline ui/pages/add-nft/add-nft.js:42]:
```suggestion
  ButtonIconSize,
```

**Review by brad-decker — DISMISSED** [review]

File: ui/pages/send/send-content/add-recipient/domain-input.component.js [hunk 2]
```diff
@@ -130,11 +130,11 @@ export default class DomainInput extends Component {
                 )}
               </div>
               <ButtonIcon
-                iconName={ICON_NAMES.CLOSE}
+                iconName={IconName.Close}
                 ariaLabel={t('close')}
                 onClick={this.props.onReset}
                 className="ens-input__wrapper__action-icon-button"
-                size={ICON_SIZES.SM}
+                size={IconSize.Sm}
```
Comment (NidhiKJha) [inline ui/pages/send/send-content/add-recipient/domain-input.component.js:137]:
```suggestion
                size={ButtonIconSize.Sm}
```

File: ui/pages/send/send-content/add-recipient/domain-input.component.js [hunk 3]
```diff
@@ -9,11 +9,11 @@ import {
...
 } from '../../../../../shared/modules/hexstring-utils';
 import {
+  ButtonIcon,
   Icon,
-  ICON_NAMES,
-  ICON_SIZES,
-} from '../../../../components/component-library/icon/deprecated';
-import { ButtonIcon } from '../../../../components/component-library/button-icon/deprecated';
+  IconName,
+  IconSize,
```
Comment (NidhiKJha) [inline ui/pages/send/send-content/add-recipient/domain-input.component.js:15]:
```suggestion
  ButtonIconSize,
```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
LGTM! Just some small nits

File: ui/pages/add-nft/add-nft.js [hunk 4]
```diff
@@ -163,8 +163,8 @@ export default function AddNft() {
                     </Typography>
                     <ButtonIcon
                       className="add-nft__close"
-                      iconName={ICON_NAMES.CLOSE}
-                      size={ICON_SIZES.SM}
+                      iconName={IconName.Close}
+                      size={IconSize.Sm}
```
Comment (NidhiKJha) [inline ui/pages/add-nft/add-nft.js:167]:
```suggestion
                      size={ButtonIconSize.Sm}
```

File: ui/pages/settings/contact-list-tab/contact-list-tab.component.js [hunk 5]
```diff
@@ -79,10 +79,10 @@ export default class ContactListTab extends Component {
       <div className="address-book__container">
         <div>
           <Icon
-            name={ICON_NAMES.BOOK}
+            name={IconName.Book}
             color={Color.iconMuted}
```
Comment (georgewrmarshall) [inline ui/pages/settings/contact-list-tab/contact-list-tab.component.js:83]:
```suggestion
            color={IconColor.iconMuted}
```

File: ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js [hunk 6]
```diff
@@ -93,10 +91,10 @@ const SnapListTab = () => {
             alignItems={AlignItems.center}
           >
             <Icon
-              name={ICON_NAMES.SNAPS}
+              name={IconName.Snaps}
               color={Color.iconMuted}
```
Comment (georgewrmarshall) [inline ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js:95]:
```suggestion
              color={IconColor.iconMuted}
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good couple small suggestions

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**